kylin/src/main/java/org/apache/zeppelin/kylin/KylinInterpreter.java

```diff
@@ -0,0 +1,197 @@
+
+
+  static final String KYLIN_QUERY_API_URL = "kylin.api.url";
+  static final String KYLIN_USERNAME = "kylin.api.user";
+  static final String KYLIN_PASSWORD = "kylin.api.password";
+  static final String KYLIN_QUERY_PROJECT = "kylin.query.project";
+  static final String KYLIN_QUERY_OFFSET = "kylin.query.offset";
+  static final String KYLIN_QUERY_LIMIT = "kylin.query.limit";
+  static final String KYLIN_QUERY_ACCEPT_PARTIAL = "kylin.query.ispartial";
+  static final Pattern KYLIN_TABLE_FORMAT_REGEX_LABEL = Pattern.compile("\"label\":\"(.*?)\"");
+  static final Pattern KYLIN_TABLE_FORMAT_REGEX = Pattern.compile("\"results\":\\[\\[\"(.*?)\"]]");
+
+  static {
+    Interpreter.register(
+        "kylin",
+        "kylin",
+        KylinInterpreter.class.getName(),
+        new InterpreterPropertyBuilder()
+            .add(KYLIN_USERNAME, "ADMIN", "username for kylin user")
+            .add(KYLIN_PASSWORD, "KYLIN", "password for kylin user")
+            .add(KYLIN_QUERY_API_URL, "http://<host>:<port>/kylin/api/query", "Kylin API.")
+            .add(KYLIN_QUERY_PROJECT, "default", "kylin paroject name")
+            .add(KYLIN_QUERY_OFFSET, "0", "kylin query offset")
+            .add(KYLIN_QUERY_LIMIT, "5000", "kylin query limit")
+            .add(KYLIN_QUERY_ACCEPT_PARTIAL, "true", "The kylin query partial flag").build());
+  }
+
+
+
+
+  public KylinInterpreter(Properties property) {
+    super(property);
+  }
+  @Override
+  public void open() {
+
+  }
+
+  @Override
+  public void close() {
+
+  }
+
+  @Override
+  public InterpreterResult interpret(String st, InterpreterContext context) {
+    try {
+      return executeQuery(st);
+    } catch (IOException e) {
+      logger.info("failed to query data in kylin ", e.getMessage());
+      return new InterpreterResult(InterpreterResult.Code.ERROR, e.getMessage());
+    }
+  }
+
+  @Override
+  public void cancel(InterpreterContext context) {
+
+  }
+
+  @Override
+  public FormType getFormType() {
+    return FormType.SIMPLE;
+  }
+
+  @Override
+  public int getProgress(InterpreterContext context) {
+    return 0;
+  }
+
+  @Override
+  public List<String> completion(String buf, int cursor) {
+    return null;
+  }
+
+  private InterpreterResult executeQuery(String sql) throws IOException {
+
+    logger.info("project" + getProperty(KYLIN_QUERY_PROJECT));
+    logger.info("sql:" + sql);
+    logger.info("acceptPartial:" + getProperty(KYLIN_QUERY_ACCEPT_PARTIAL));
+    logger.info("limit:" + getProperty(KYLIN_QUERY_LIMIT));
+    logger.info("offset:" + getProperty(KYLIN_QUERY_OFFSET));
+    StringBuilder msg = null;
+    byte[] encodeBytes = Base64.encodeBase64(new String(getProperty(KYLIN_USERNAME)
+        + ":" + getProperty(KYLIN_PASSWORD)).getBytes());
+    HttpClient httpClient = HttpClientBuilder.create().build();
+
+    String postContent = new String("{\"project\":" + "\"" + getProperty(KYLIN_QUERY_PROJECT) + "\""
```

Comment:
Store `getProperty(KYLIN_QUERY_PROJECT)` in a local variable. It is read more than once in this method.